idseq_bench/scoring/metrics.py

```diff
@@ -0,0 +1,35 @@
+from collections import defaultdict
+from sklearn.metrics import auc, precision_recall_curve
+import numpy as np
+
+def adjusted_aupr(y_true, y_score, force_monotonic=False):
+  # Adapted from https://github.com/yesimon/metax_bakeoff_2019/blob/master/plotting/Metagenomics%20Bench.ipynb
+  original_precision, recall, thresholds = precision_recall_curve(y_true, y_score)
+
+  precision = original_precision
+  if force_monotonic:
+    # adjusts precision per each recall valu on the curve
+    # it guarantees that the curve is monotonic decreasing
+    precision_max_per_recall = defaultdict(float)
+    for r, p in zip(recall, original_precision):
+      precision_max_per_recall[r] = max(precision_max_per_recall[r], p)
+    adjusted_precision = []
+    for r, p in zip(recall, original_precision):
+      adjusted_precision.append(precision_max_per_recall[r])
+    precision = adjusted_precision
+
+  # force start at zero
+  if thresholds[0] == 0:
+    precision[0] = 0
+    recall[0] = recall[1]
+    recall = np.insert(recall, 0, 1)
+    precision = np.insert(precision, 0, 0)
+
+  aupr = auc(recall, precision)
+  return {k: v for k, v in {
+    "aupr": aupr,
+    "original_precision": original_precision if force_monotonic else None,
```

Comment:
What is the benefit of conditionally setting this to `None`? 
Does it make more sense to always set to `original_precision`, then the `original_precision` and `precision` values would just be equal for cases where `force_monotonic=False`?

Reply:
By setting to None I can remove the value below. So, original precision is not returned in that case.
Would you prefer to return the the same values?

Reply:
I see! I didn't catch the implication of the removal below. I like the current implementation to remove that value if it's irrelevant - makes interpreting the results more clear if you didn't know the specifications to run the benchmark formula. Thanks!